### ISSUE TITLE
failed with exit code 133.  (rc=34048)

### ISSUE THREAD

**Issue by bifidoman** [marbl/canu#560]
Hi

I just installed canu and tried the test data in the quick start guide. I get this error:
```
----------------------------------------
ERROR:
ERROR:  Failed with exit code 133.  (rc=34048)
ERROR:
================================================================================
Don't panic, but a mostly harmless error occurred and Canu stopped.

  Reason: image not found failed with:
  gatekeeper store exists, but contains no reads
===============================================================================
```

I found a previous issue in this and it stated install it form the source code. I did that, and I got the same message again

I run it on a mac. I do not have problems running it on a linux, but I just got a new Mac which is much faster

any suggestions welcome

thanks
Marc

**Comment by brianwalenz** [comment]
'Reason: image not found' isn't a canu error.  This makes me think it didn't compile/install correctly.  Can you paste in more of the screen output, and whatever is in any error file in the correction/ directory or subdirectory?  *.gkpStore.err is probably the important one.

Just above this error is the gatekeeperCreate command being run.  Can you run either this command (with no options supplied, it'll just report a usage statement) or the command with options?  Leave off the redirection to the error file, so something like:
```
    /work/canu/Darwin-amd64/bin/gatekeeperCreate \
      -minlength 1000 \
      -o ./test.gkpStore.BUILDING \
      ./test.gkpStore.gkp
```
This expects to be run from inside the correction/ subdirectory, where the *gkpStore.gkp file exists.

**Comment by bifidoman** [comment]
Hi


thanks for the quick response

Here is the TW output:
canu \
>  -p ecoli -d ecoli-auto \
>  genomeSize=4.8m \
>  -pacbio-raw p6.25x.fastq
--   Reason: image not found
-- Detected Java(TM) Runtime Environment '1.8.0_131' (from 'java').
-- Detected gnuplot version '5.0 patchlevel 6' (from 'gnuplot') and image format 'png'.
-- Detected 8 CPUs and 32 gigabytes of memory.
-- No grid engine detected, grid disabled.
--
-- Run   2 jobs concurrently using    8 GB and   4 CPUs for stage 'meryl'.
-- Run   1 job  concurrently using    6 GB and   8 CPUs for stage 'mhap (cor)'.
-- Run   1 job  concurrently using    8 GB and   8 CPUs for stage 'overlapper (obt)'.
-- Run   1 job  concurrently using    8 GB and   8 CPUs for stage 'overlapper (utg)'.
-- Run   4 jobs concurrently using    8 GB and   2 CPUs for stage 'falcon_sense'.
-- Run   8 jobs concurrently using    4 GB and   1 CPU  for stage 'ovStore bucketizer'.
-- Run   8 jobs concurrently using    8 GB and   1 CPU  for stage 'ovStore sorting'.
-- Run   2 jobs concurrently using    2 GB and   4 CPUs for stage 'read error detection'.
-- Run   8 jobs concurrently using    1 GB and   1 CPU  for stage 'overlap error adjustment'.
-- Run   2 jobs concurrently using   16 GB and   4 CPUs for stage 'bogart'.
-- Run   8 jobs concurrently using    4 GB and   1 CPU  for stage 'GFA alignment and processing'.
-- Run   2 jobs concurrently using   16 GB and   4 CPUs for stage 'consensus'.
--
-- Generating assembly 'ecoli' in '/Users/generaluser/Documents/software/canu/canu-1.5/Darwin-amd64/bin/ecoli-auto'
--
-- Parameters:
--
--  genomeSize        4800000
--
--  Overlap Generation Limits:
--    corOvlErrorRate 0.2400 ( 24.00%)
--    obtOvlErrorRate 0.0450 (  4.50%)
--    utgOvlErrorRate 0.0450 (  4.50%)
--
--  Overlap Processing Limits:
--    corErrorRate    0.3000 ( 30.00%)
--    obtErrorRate    0.0450 (  4.50%)
--    utgErrorRate    0.0450 (  4.50%)
--    cnsErrorRate    0.0450 (  4.50%)
--
--
-- BEGIN CORRECTION
--
----------------------------------------
-- Starting command on Mon Jul 24 08:57:21 2017 with 1077.751 GB free disk space

    cd correction
    /usr/local/Cellar/canu/1.5/Darwin-amd64/bin/gatekeeperCreate \
      -minlength 1000 \
      -o ./ecoli.gkpStore.BUILDING \
      ./ecoli.gkpStore.gkp \
    > ./ecoli.gkpStore.BUILDING.err 2>&1
sh: line 4: 28360 Trace/BPT trap: 5       /usr/local/Cellar/canu/1.5/Darwin-amd64/bin/gatekeeperCreate -minlength 1000 -o ./ecoli.gkpStore.BUILDING ./ecoli.gkpStore.gkp > ./ecoli.gkpStore.BUILDING.err 2>&1

-- Finished on Mon Jul 24 08:57:21 2017 (lickety-split) with 1077.751 GB free disk space
----------------------------------------
ERROR:
ERROR:  Failed with exit code 133.  (rc=34048)
ERROR:
================================================================================
Don't panic, but a mostly harmless error occurred and Canu stopped.

  Reason: image not found failed with:
  gatekeeper store exists, but contains no reads
-----------------------------------------------------------------------------------------------



this is the .gkpStore.err file:
--------------------------------------------------------------------------
###
###  Reading options from '/usr/local/Cellar/canu/1.5/Darwin-amd64/bin/canu.defaults'
###

# Add site specific options (for setting up Grid or limiting memory/threads) here.

###
###  Reading options from the command line.
###

genomeSize=4.8m
-------------------------------------------------------------------



this is the .gkpStore.gkp file:
------------------------------------------------------------------
########################################
#  pacbio-raw: /Users/generaluser/Documents/software/canu/canu-1.5/Darwin-amd64/bin/p6.25x.fastq
#
name   p6.25x
preset pacbio-raw
/Users/generaluser/Documents/software/canu/canu-1.5/Darwin-amd64/bin/p6.25x.fastq
-----------------------------------------------------------------



Ad here is the command run

**Comment by bifidoman** [comment]
Ad here is the command run:


bt141-244:correction generaluser$ ls
ecoli.gkpStore.err	ecoli.gkpStore.gkp
bt141-244:correction generaluser$ gatekeepercreate
usage: gatekeepercreate [...] -o gkpStore
  -o gkpStore         create this gkpStore
  -a gkpStore         append to this gkpStore
  
  -minlength L        discard reads shorter than L
  
  
ERROR: no gkpStore (-o) supplied.
ERROR: no input files supplied.



So, the Gatekeeper runs, but does not find the gkpstore file?

looks like a Mac problem, I am searching for solutions
However, hope you can help. 
thanks

Marc

**Comment by brianwalenz** [comment]
Well, my two guesses aren't true, so we're back to 'try this, try that'.

Run this:
```
    cd correction
    /usr/local/Cellar/canu/1.5/Darwin-amd64/bin/gatekeeperCreate \
      -minlength 1000 \
      -o ./test.gkpStore \
      ./ecoli.gkpStore.gkp \
```
It's just the command that fails inside canu, with a different output name.  What's it write to the screen?  What's inside directory 'test.gkpStore' (ls -l test.gkpStore)?  What's in test.gkpStore.err and test.gkpStore/errorLog?

This looks like a very old version of canu, so try updating to the latest release (1.5) or the latest unstable from github.  Are you installing binaries or compiling it yourself?

**Comment by bifidoman** [comment]
Output look like this:


dyld: Library not loaded: /usr/local/opt/gcc/lib/gcc/6/libgomp.1.dylib
  Referenced from: /usr/local/Cellar/canu/1.5/Darwin-amd64/bin/gatekeeperCreate
  Reason: image not found
Trace/BPT trap: 5
bt141-244:correction generaluser$ 

I am running version 1.5

I will reinstall. Maybe...

**Comment by skoren** [comment]
The error indicates you're missing the OpenMP libraries which the canu binary distribution is built with. You can download and build from source (so not the file that has Darwin or Linux in the name) but you won't get any of the parallelism in the code.  I suggest installing GCC using any of the mac package managers and then build locally.

**Comment by bifidoman** [comment]
It works!

I checked already, GCC 7.1.0 is installed, and I already tried to build from Canu source (because that was mentioned in a previous issue on error code 133. Did not work
However, than I found the following statement here: https://github.com/torch/distro/issues/46
"the homebrew gcc is probably the issue. I'd suggest using the default clang with updated XCode. Just tried it on a fresh el capitan machine (which had no torch installed ever), no issues."

So, I uninstalled all GCC versions and than reinstalled it, but NOT with homebrew. Than I reinstalled Canu and it is running now (Hope all other GCC dependencies still do..).

anyway, tanks for your help. Your comments were very useful to identify the problem
